### ISSUE TITLE
Method to check what formats are supported

### ISSUE THREAD

**Issue by GuySartorelli** [Intervention/image#1326]
**Describe the feature you'd like**
A method that I can call which tells me either:
- All supported formats, or
- Whether a specific format is supported

e.g.
```php
use Intervention\Image\ImageManager;
use Intervention\Image\Drivers\Gd\Driver as GdDriver;

$manager = new ImageManager(new GdDriver());

$isSupported = $manager->supports('png'); // bool
$allSupported = $manager->getSupportedFormats(); // array
```

Imagick has [`Imagick::queryFormats()`](https://www.php.net/manual/en/imagick.queryformats.php) which provides this information, and GD has the [`imagetypes()` function](https://www.php.net/manual/en/function.imagetypes.php).

Even if it's not dynamically checking based on the above, intervention image does surely know what it can and can't support without relying on throwing exceptions?

**Is your feature request related to a problem? Please describe.**
Right now I either have to call those methods directly or have a hardcoded list of formats. I'm currently writing an abstract file converter interface, and using intervention as a use case to make sure the API is fit for purpose. It's very frustrating that I need to define or find a list of supported formats on my own... this feels very much like something intervention image should be handling for me.

**Comment by olivervogel** [comment]
Sounds like a good idea. The library knows which formats are supported in the best case. However, the decisive factor is with which support the extension (GD or Imagick) was built. The functions mentioned can be used to check this.

I will look into this.

**Comment by olivervogel** [comment]
This feature is available as of version `3.6.0`. See: https://image.intervention.io/v3/introduction/formats#check-the-support-for-image-formats
